### PR TITLE
ERSSUP-54590 - OAS documentation for ERSSUP-54152: APIM User calls e-RS Endpoint to Upload Attachment File to Document Store

### PR DESCRIPTION
**Pull request by nhsd-hubert-raich**

### CODE REVIEW

File: sandbox/src/routes/uploadFileToDocumentStore.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  /**
+   * Sandbox implementation for uploadFileToDocumentStore endpoint
+   */
+  {
+    method: 'POST',
+    path: '/STU3/v1/Binary',
+    config: {
+      payload: {
+        parse: false,
+        allow: ['text/plain', 'text/html', 'application/pdf', 'text/xml', 'text/rtf',
```
Comment (nhsd-mark-warren) [inline sandbox/src/routes/uploadFileToDocumentStore.js:13]: Is it worth explaining where this list comes from in the comment?

File: specification/e-referrals-service-api.yaml [hunk 2]
```diff
@@ -537,9 +537,133 @@ paths:
...
+        You can only upload file formats that are supported by e-RS. Currently, the supported file formats are:
+
+        | File type          | File extension(s)           |
+        | ------------------ | --------------------------- |
+        | Plain Text         | `.txt`                      |
+        | HTML Text          | `.html,` `.htm`             |
+        | PDF                | `.pdf`                      |
+        | XML Text           | `.xml`                      |
+        | RTF Text           | `.rtf`, `.rtx`              |
+        | Basic Audio        | `.pdf`                      |
```
Comment (nhsd-mark-warren) [inline specification/e-referrals-service-api.yaml:564]: PDF -> basic audio .. really ?

File: specification/e-referrals-service-api.yaml [hunk 3]
```diff
@@ -537,9 +537,133 @@ paths:
...
+        | MPEG audio layer 3 | `.mp3`                      |
+        | PNG Image          | `.png`                      | 
+        | GIF Image          | `.gif`                      |
+        | JPEG Image         | `.jpg,` `.jpe`, `.jpeg`     |
+        | TIFF Image         | `.tif`, `.tiff`             |
+        | MPEG Video         | `.mpg`, `.mpeg`, `.mpe`     |
+        | MSWORD             | `.doc`, `.docx`             |
+
+        Currently, the maximum file size supported by e-RS is 5MB.
```
Comment (nhsd-mark-warren) [inline specification/e-referrals-service-api.yaml:573]: I guess you can't document it here, but out of interest where is the "source of truth" for this max file size?

File: specification/e-referrals-service-api.yaml [hunk 4]
```diff
@@ -537,9 +537,133 @@ paths:
...
+
+        As an authenticated e-RS user
+
+        I need to upload a referral letter or other clinical information file to e-RS
+
+        So that I can associate the file with an existing referral.
+
+        ## Important notes:
+
+        This endpoint **only** performs the file upload. The uploaded file is **not** associated with a referral until the file has been associated with a referral, for example using [[HYPERLINK_A012]].
```
Comment (nhsd-mark-warren) [inline specification/e-referrals-service-api.yaml:585]: This sentence seems a bit clunky

File: specification/e-referrals-service-api.yaml [hunk 5]
```diff
@@ -537,9 +537,133 @@ paths:
...
+              $ref: '#/components/schemas/StringBinarySchema'
+          application/pdf:
+            schema:
+              $ref: '#/components/schemas/StringBinarySchema'
+          text/xml:
+            schema:
+              $ref: '#/components/schemas/StringBinarySchema'
+          text/rtf:
+            schema:
+              $ref: '#/components/schemas/StringBinarySchema'
```
Comment (nhsd-mark-warren) [inline specification/e-referrals-service-api.yaml:620]: In the case of RTF, are there any restrictions on the character encoding that must be used?

**Review by nhsd-mark-warren — APPROVED** [review]

File: specification/e-referrals-service-api.yaml (hunk 5 above)
Comment (nhsd-hubert-raich) [reply specification/e-referrals-service-api.yaml:620]: @adamoldfield-nhs  do you happen to know the answer to this question ?

File: specification/e-referrals-service-api.yaml (hunk 2 above)
Comment (nhsd-hubert-raich) [reply specification/e-referrals-service-api.yaml:564]: Seem to be a copy&paste mistake. I'll fix it

File: specification/e-referrals-service-api.yaml (hunk 3 above)
Comment (nhsd-hubert-raich) [reply specification/e-referrals-service-api.yaml:573]: I'm not sure where it's defined but there are number of components at play here. There is another stream of work that is going to concentrate on increasing this limit.

File: specification/e-referrals-service-api.yaml (hunk 4 above)
Comment (nhsd-hubert-raich) [reply specification/e-referrals-service-api.yaml:585]: We can't change the text in overview section at the moment, it was already refined and approved by BA's and Business. It'd have to go on another loop if we were to update it.  There is going to be a final review of full content before this gets published.

**Pull request merged**